src/main/java/net/sf/mzmine/modules/tools/msmsspectramerge/MsMsSpectraMerge.java

```diff
@@ -0,0 +1,383 @@
+package net.sf.mzmine.modules.tools.msmsspectramerge;
+
+import com.google.common.collect.Range;
+import net.sf.mzmine.datamodel.*;
+import net.sf.mzmine.main.MZmineCore;
+import net.sf.mzmine.modules.MZmineModule;
+import net.sf.mzmine.parameters.ParameterSet;
+import net.sf.mzmine.util.scans.ScanUtils;
+import org.apache.commons.math3.special.Erf;
+import org.slf4j.LoggerFactory;
+
+import javax.annotation.Nonnull;
+import javax.annotation.Nullable;
+import java.util.*;
+import java.util.stream.Collectors;
+
+/**
+ * Module for merging MS/MS spectra. Merging is performed by:
+ * 1. first selecting all consecutive MS/MS and the surrounding MS1 scans
+ * 2. scoring each such list of consecutive MS/MS by the intensity of the precursor peak in the surrounding MS1 divided
+ * by the intensity of chimeric peaks around them
+ * 3. removing MS/MS with considerably low scoring (<20% of best score)
+ * 4. start with MS/MS with best score. Merge iteratively all spectra into this MS/MS which have a cosine score
+ * above the user given threshold.
+ *
+ * The reasoning behind 4. is that the method will first select a MS/MS with, hopefully, low chimeric contamination.
+ * Afterwards it only merges MS/MS into this spectra which do not deviate too much (e.g. have low chimerics, too)
+ *
+ * 5. Merging all merged spectra within a sample belonging to the same feature using the same routine
+ * 6. Merging all merged spectra across samples belonging to the same feature using the same routine
+ * 7. removing peaks from merged spectra which are not consistent across the merged spectra
+ */
+public class MsMsSpectraMerge implements MZmineModule {
```

Comment:
Please also add this module to the MZmineModulesList class, so its configuration is saved upon exit.